vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +89,339 @@ def process_shapes(
         with the identified shape or {"shape": None} if the contour does not match any.
     """
-    retval_boxes: list[tuple[int, int, int, int]] = [cv2.boundingRect(cnt) for cnt in contours]
-    verts_lst: list[bbox.Vertices] = [
-        bbox.tlwh_to_vertices(retval_box[0], retval_box[1], retval_box[2], retval_box[3])
-        for retval_box in retval_boxes
-    ]
-    boxes: list[bbox.BoundingBox] = [
-        bbox.BoundingBox(verts, bbox.ObjectType.STD_OBJECT) for verts in verts_lst
-    ]
-
-    shape_boxes: list[bbox.BoundingBox] = []
-
-    idx: int
-    hier: npt.NDArray[npt.Shape["4"], npt.IntC]
-    box: bbox.BoundingBox
-    # for each contour (and corresponding elements in hierarchy array and box list)
-    for idx, (hier, box) in enumerate(zip(hierarchy[0], boxes)):
-        classification: chars.ODLCShape | None
-
-        # as long as the contour is not inside another contour that has been classified as a shape
-        if (
-            hier[3] == -1
-            or not boxes[hier[3]].attributes
-            or boxes[hier[3]].attributes["shape"] is None
-        ) and get_child_amt(idx, hierarchy) <= MAX_CHILD_AMT:
-            classification = classify_shape(contours[idx], image_dims)
-        else:
-            # if the contour is inside an identified ODLC shape then it should not be recognized
-            classification = None
-
-        if classification is not None:
-            box.set_attribute("shape", classification)
-            shape_boxes.append(box)
-
-    return shape_boxes
-
-
-def classify_shape(
-    contour: consts.Contour,
-    image_dims: tuple[int, int],
-    approx_contour: consts.Contour | None = None,
-) -> chars.ODLCShape | None:
+    bbox_list: List[bbox]
+    for contour in contours:
+        Shape_Type: chars.ODLCShape | None = classify_shape(contour)
+
+        if not Shape_Type == None:
+            # sorted_points = [sorted([point for point in sorted_points if point[0] == x], key=lambda y: y[1]) for x, _ in sorted_points]
+            min_y: int = contour[0][0][0]
+            max_y: int = contour[0][0][0]
+            min_x: int = contour[0][0][1]
+            max_x: int = contour[0][0][1]
+            vertices: consts.Corners
+
+            for list in contour:
+                point: Tuple[int, int] = list[0]
+                y, x = list
+                if y > max_y:
+                    max_y = y
+                if y < min_y:
+                    min_y = y
+                if x > max_x:
+                    max_x = x
+                if x < min_x:
+                    min_x = x
+            height: int = max_y - min_y
+            width: int = max_x - min_x
+            vertices = getVertices(min_x, max_y, width, height)
+
+        bounding_box: bbox
+        bbox.__init__(bounding_box, vertices, ObjectType.STD_OBJECT, None)
+        bbox_list.append(bounding_box)
+    return bbox_list
+
+
+def classify_shape(contour: consts.Contour) -> Union[chars.ODLCShape, None]:
```

Comment:
Union types aren't used in Python 3.10. Instead,
```suggestion
def classify_shape(contour: consts.Contour) -> chars.ODLCShape | None:
```